pkg/userdata/rhel/provider.go

```diff
@@ -204,7 +204,9 @@ write_files:
 {{- /*  https://bugs.launchpad.net/cloud-init/+bug/1662542 */}}
     hostnamectl set-hostname {{ .MachineSpec.Name }}
     {{ end }}
-
+    {{ if eq .CloudProviderName "azure" }}
+    yum update -y --disablerepo='*' --enablerepo='*microsoft*'
```

Comment:
Isn't this should be gated under `{{- if .OSConfig.DistUpgradeOnBoot }}`?

Reply:
I mean, in addition to `if eq .CloudProviderName "azure"`

Reply:
Why? the reason why I disabled those repos due to some openssl certificates that are not consistent by some mirrors. After upgrading to use the Gold Access images for azure and then enabled repos from MS mirrors all worked.

Reply:
Since we use GA image, does it mean this line can be removed then?

Reply:
No, because either cases whether you have a GA image or a custom one, rhui client would still try to access the repos. Thus this will end up with the same situation where rhui client certificates are expired and it would fail on all mirrors. I was expecting that updating `yum update` alone would actually update the certificate as well, however this is not the case for RHEL 8.x